SFML/SFML
# Simplify CMake build by removing source_group and headers as source files

## Description

Strictly speaking CMake puts no requirement on using `source_group` or listing headers as source files. As I've heard, both of these practices are conveniences to Visual Studio users. Having never used Visual Studio myself I struggle to understand how much of a convenience this is. I'm submitting this PR to get more feedback on what VS users want and to demonstrate exactly how much code can be simplified if we're willing to stop doing these things.

## Tasks


## How to test this PR?

Build this PR in Visual Studio and try to observe any difference in your workflow, particularly if this PR makes SFML development more difficult or annoying. Share any specific observations you make. Mention what version of Visual Studio you are using.
